library/core/src/ptr/mod.rs

```diff
@@ -880,7 +882,8 @@ pub unsafe fn read_unaligned<T>(src: *const T) -> T {
 /// ```
 #[inline]
 #[stable(feature = "rust1", since = "1.0.0")]
-pub unsafe fn write<T>(dst: *mut T, src: T) {
+#[rustc_const_unstable(feature = "const_ptr_write", issue = "none")]
+pub const unsafe fn write<T>(dst: *mut T, src: T) {
```

Comment:
See [this Zulip thread](https://rust-lang.zulipchat.com/#narrow/stream/131828-t-compiler/topic/ptr.3A.3Aread.20vs.20ptr.3A.3Awrite) -- I think we can implement this without the intrinsic, and that seems more consistent with `ptr::read`.

Reply:
Ok. Should I make `move_val_init` const then?

Reply:
Maybe leave out `ptr::write` for now until that Zulip thread reaches a conclusion.

Reply:
Sure.

Possibly relevant to the Zulip discussion? - The compiler does not seem too happy about the reference to `src` in `const fn` context

```rust
error[E0492]: cannot borrow a constant which may contain interior mutability, create a static instead
   --> library/core/src/ptr/mod.rs:890:29
    |
890 |         copy_nonoverlapping(&src as *const T as *const u8, dst as *mut u8, mem::size_of::<T>());
    |                             ^^^^
```

Reply:
Ah... that's another issue that @oli-obk [recently mentioned](https://github.com/rust-lang/rust/issues/79597#issuecomment-748593118).

Yet another reason to maybe hold off on `ptr::write` for now.^^